ably/paginated_result.go

```diff
@@ -59,7 +62,15 @@ func newPaginatedResult(typ reflect.Type, path string, params *PaginateParams,
 	p.path = builtPath
 	p.links = resp.Header["Link"]
 	v := reflect.New(p.typ)
-	if err = json.NewDecoder(resp.Body).Decode(v.Interface()); err != nil {
+	switch typ := strip(resp.Header.Get("Content-Type"), ';'); typ {
```

Comment:
Use [mime.ParseMediaType](https://golang.org/pkg/mime/#ParseMediaType) (and remove the `strip` function).